### PR TITLE
Disable warning about Manifest v2 deprecation

### PR DESCRIPTION
**Pull request by antonok-edm**
Resolves https://github.com/brave/brave-browser/issues/26207

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Download a Chromium build of a Manifest V2 extension. For example, download `uBlock0_1.44.5b17.chromium.zip` from [here](https://github.com/gorhill/uBlock/releases/tag/1.44.5b17)
2. Unzip the downloaded extension
3. Visit brave://extensions
4. Ensure `Developer mode` is enabled (there is a toggle in the top-right corner)
5. Click the `Load unpacked` button at the top-left corner, navigate to the unzipped extension directory, and choose `Select folder`.
6. The extension should be installed. Ensure that there is no `Errors` button to the right of the `Details` and `Remove` buttons on the newly installed extension.

### CODE REVIEW

File: test/filters/browser_tests.filter [hunk 1]
```diff
@@ -9,6 +9,13 @@
 # display an additional confirmation infobar for P3A
 -All/PageSpecificSiteDataDialogBrowserTest.*/*
 
+# These tests fail because they expect an additional Manifest V2 deprecation
```
Comment (bridiver) [inline test/filters/browser_tests.filter:12]: we should add similar Brave tests to check to ensure Manifest V2 is not deprecated

Comment (antonok-edm) [reply test/filters/browser_tests.filter:12]: added

File: browser/brave_browser_main_parts.cc [hunk 2]
```diff
@@ -158,6 +159,8 @@ void BraveBrowserMainParts::PreShutdown() {
 
 void BraveBrowserMainParts::PreProfileInit() {
   ChromeBrowserMainParts::PreProfileInit();
+  extensions::Extension::
```
Comment (bridiver) [inline browser/brave_browser_main_parts.cc:162]: please add comment explaining what this is doing

Comment (antonok-edm) [reply browser/brave_browser_main_parts.cc:164]: done

File: browser/brave_browser_main_extra_parts.cc [hunk 3]
```diff
@@ -54,6 +55,14 @@ BraveBrowserMainExtraParts::BraveBrowserMainExtraParts() = default;
 
 BraveBrowserMainExtraParts::~BraveBrowserMainExtraParts() = default;
 
+void BraveBrowserMainExtraParts::PreProfileInit() {
+#if !BUILDFLAG(IS_ANDROID)
```
Comment (bridiver) [inline browser/brave_browser_main_extra_parts.cc:59]: this should be `ENABLE_EXTENSIONS`

**Review by bridiver — APPROVED** [review]

**Pull request merged**